### PR TITLE
Loggers

### PR DESCRIPTION
**Pull request by AnthonySteele**
Make logging more flexible. `--logfile somefile.log` will append output to that file.

WIP towards #331 , and generally making the local inspect output nicer and more flexible, and reworking the report output

### CODE REVIEW

File: NuKeeper.Inspection/Logging/IConfigureLogger.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+{
+    public enum LogDestination
+    {
+        Console,
+        File
+    }
+
+    public interface IConfigureLogger
+    {
+        void Initialise(LogLevel logLevel, string filePath);
```
Comment (skolima) [inline NuKeeper.Inspection/Logging/IConfigureLogger.cs]: Uhm, that file is a bit odd. Enum isn't related to the interface, and the interface is mixing log levels and file logging.

File: NuKeeper.Inspection/Logging/IInternalLogger.cs [hunk 2]
```diff
@@ -0,0 +1,10 @@
+using System;
+
+namespace NuKeeper.Inspection.Logging
+{
+    public interface IInternalLogger
+    {
+        void Error(string message, Exception ex = null);
```
Comment (skolima) [inline NuKeeper.Inspection/Logging/IInternalLogger.cs]: Personal bias against optional parameters, but: this makes it way easier to forget to include the actual exception when logging. I'd vote for explicit `null` in those cases.

Comment (AnthonySteele) [reply NuKeeper.Inspection/Logging/IInternalLogger.cs]: Yep, the default is no longer needed on the internal logger.

**Comment by skolima** [comment]
`--logfile` seems to be the standard switch https://docs.microsoft.com/en-us/visualstudio/msbuild/msbuild-command-line-reference?view=vs-2017

**Pull request merged**